src/Compilers/CSharp/Portable/Symbols/SubstitutedMethodSymbol.cs

```diff
@@ -360,19 +360,30 @@ internal override int CalculateLocalSyntaxOffset(int localPosition, SyntaxTree l
         private int ComputeHashCode()
         {
             int code = this.OriginalDefinition.GetHashCode();
-            code = Hash.Combine(this.ContainingType, code);
-
-            // Unconstructed method may contain alpha-renamed type parameters while
-            // may still be considered equal, we do not want to give different hashcode to such types.
-            //
-            // Example:
-            //   Having original method A<U>.Goo<V>() we create two _unconstructed_ methods
-            //    A<int>.Goo<V'>
-            //    A<int>.Goo<V">     
-            //  Note that V' and V" are type parameters substituted via alpha-renaming of original V
-            //  These are different objects, but represent the same "type parameter at index 1"
-            //
-            //  In short - we are not interested in the type arguments of unconstructed methods.
+
+            // If the containing type of the of the original definition is the same as our containing type
+            // it's possible that we will compare equal to the original definition under certain conditions 
+            // (e.g, ignoring nullability) and want to retain the same hashcode. As such, consider only
+            // the the original definition for the hashcode when we know equality is possible
+            var containingHashCode = _containingType.GetHashCode();
+            if (containingHashCode == this.OriginalDefinition.ContainingType.GetHashCode())
+            {
+                return code;
```

Comment:
>return code [](start = 16, length = 11)

Any chance this returns `0`?

Reply:
But then we won't match the original definition hash code, which we have to do. If in the rare case it *is* `0` we'll just call CalculateHashCode each time rather than caching it, but it's pretty cheap as we are just going to call into the original definition and return.